elastic/kibana
# [Fleet] fix policy validation

## Summary

Fixes https://github.com/elastic/kibana/issues/123704

Package policy could be submitted with an invalid form (missing required field).

it looks like the problem is that the formState is set to VALID when agent policy is updated on the form, this seems to happen even if the form has validation errors.
https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/index.tsx#L177

To test:
Add Custom Logs integration: http://localhost:5601/app/fleet/integrations/log-1.0.0/add-integration
Verify that Log file path is empty, and when clicking on the input, it shows up as required.
Verify that the form can't be submitted until the field value is set (changing agent policy should not make the form valid)

![image](https://user-images.githubusercontent.com/90178898/151575613-98f1e5e8-6700-4f39-b7b7-b59ad10afed6.png)
